python/tests/test_framework/slate/data_generation.py

```diff
@@ -0,0 +1,78 @@
+import random
+import os
+from test_helper import get_function_object
+
+script_directory = os.path.dirname(os.path.realpath(__file__))
+random.seed(10)
+
+
+def random_number_items(items):
+    num_items_to_select = random.randint(1, len(items))
+    return random.sample(items, num_items_to_select)
+
+
+def generate_slate_data(
+    num_examples,
+    num_actions,
+    reward_function,
+    logging_policy,
+    action_space,
+    num_slots=1,
+    num_context=1,
+    context_name=None,
+    slot_name=None,
```

Comment:
Probably better to remove slot_name and num_slots for now and start with supporting automatic slot naming (with inference of slots number from action_space output)